docs/extensions/services/interacting-with-files.md

```diff
@@ -0,0 +1,63 @@
+---
+description: Learn about the FilesService in Directus and how to utilize them when building extensions.
+contributors: Esther Agbaje
+---
+
+# Interacting With Files
+
+Directus exposes services that allow you to perform operations programmatically on your files.
+
+## FilesService
+
+The `FilesService` provides access to upload, import and perform other CRUD operations on files in Directus Files.
+
+### Import a File
+
+```js
+  router.post('/files', async (req, res) => {
+    const schema = await getSchema();
+    const service = new FilesService({ schema });
+    const assetKey = await service.importOne(req.body.url);
+
+    const data = await service.readOne(assetKey);
+    res.locals['payload'] = { data: data || null };
+
+    res.json(data);
+  });
+```
+
```

Comment:
Upload a file is missing

Reply:
Encountered a couple of errors while trying to implement the endpoint to upload a file- why I left it out. Also, imo, there doesn't seem to be a strong use case for uploading files via the API as most users will already have the url format of the file, which is what importing a file does.

Reply:
Don't agree - what if I want to build a panel that has a file upload functionality? Let's try and get this resolved - happy to pair with you 😄

Reply:
Because "receiving a file" on a custom endpoint is very tricky I'd recommend having an endpoint example that takes a base64 image string and uploads that to the file library.

Reply:
Ok but that's still a viable approach to document, no? Or do you think it reaches outside of the common asks people have.